### PR TITLE
Task/valgrind for feature 787

### PR DESCRIPTION
**Pull request by kzangeli**
Fixed the leaks from the 787 feature branch.
Also, some DB-cleanup in a few functests.
And, a retry if SHELL-INIT fails in the functional test harness.

### CODE REVIEW

File: test/functionalTest/testHarness.sh [hunk 1]
```diff
@@ -638,8 +638,25 @@ function runTest()
 
   if [ "$exitCode" != "0" ]
   then
-    exitFunction 11 "SHELL-INIT exited with code $exitCode" $path "($path)" "" DIE
-    return
+    # 3.2 Run the SHELL-INIT part AGAIN
```
Comment (fgalan) [inline test/functionalTest/testHarness.sh]: I would add some sentence about why re-running SHELL-INIT is need (something like "this step may be needed due to glitches that we suspect are due to one test starts before the former has ended).

File: src/lib/mongoBackend/MongoGlobal.cpp [hunk 2]
```diff
@@ -1786,26 +1788,28 @@ bool processOnChangeCondition
         if (condValues != NULL) {
             /* Check if some of the attributes in the NotifyCondition values list are in the entity.
              * Note that in this case we do a query for all the attributes, not restricted to attrV */
-            ContextElementResponseVector rawCerV;
+            ContextElementResponseVector rawCerV2;
```
Comment (fgalan) [inline src/lib/mongoBackend/MongoGlobal.cpp]: Why this has been renamed? Maybe a temporal fix that at the end is not needed? In that case, let's go back to `rawCerV`.

Comment (kzangeli) [inline src/lib/mongoBackend/MongoGlobal.cpp]: No, rawCerV cannot be used.
That name is already used in this function, a few lines before.
If you dont like rawCerV2, let's invent some other variable name ...

File: test/functionalTest/testHarness.sh (hunk 1 above)
Comment (kzangeli) [inline test/functionalTest/testHarness.sh]: ok

Comment (kzangeli) [inline test/functionalTest/testHarness.sh]: Fixed in 024a83c

File: src/lib/mongoBackend/MongoGlobal.cpp (hunk 2 above)
Comment (fgalan) [inline src/lib/mongoBackend/MongoGlobal.cpp]: We have end of using the `rawCerV` variable at line 1779 with `rawCerV.release()`. Thus, I understand that the variable is "free" and can be used again here. Is that correct?

Comment (kzangeli) [inline src/lib/mongoBackend/MongoGlobal.cpp]: I added that call to release :-)
It *can* be used. It is just not a very good idea to shadow your own variables.

If you really want to keep using that name, why redeclare it?
Let's just reuse the same variable ...

Comment (fgalan) [inline src/lib/mongoBackend/MongoGlobal.cpp]: Ah, ok, I think I get the point...

Redeclaring was wrong here. Thus, remove it, avoid shadowing and let's use the same variable (free at this point of code) for this block of code. Makes sense?

Comment (kzangeli) [inline src/lib/mongoBackend/MongoGlobal.cpp]: I am compiling that solution right now.

Comment (kzangeli) [inline src/lib/mongoBackend/MongoGlobal.cpp]: Fixed (and tested) in e618f7b

**Comment by fgalan** [comment]
LGTM

**Pull request merged**